### DISCUSSION TITLE
trainer.fit()

### DISCUSSION THREAD

**Discussion by Libardo1** [awslabs/gluonts#2027] [General]
Writing forecasting models in GluonTS with PyTorch

trainer = pl.Trainer(max_epochs=10, gpus=-1 if torch.cuda.is_available() else None)
trainer.fit(net, train_dataloader=data_loader)

error::
INFO:pytorch_lightning.utilities.rank_zero:GPU available: True, used: True
INFO:pytorch_lightning.utilities.rank_zero:TPU available: False, using: 0 TPU cores
INFO:pytorch_lightning.utilities.rank_zero:IPU available: False, using: 0 IPUs
INFO:pytorch_lightning.utilities.rank_zero:HPU available: False, using: 0 HPUs
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-55-2d8022b11b52>](https://localhost:8080/#) in <module>()
      1 trainer = pl.Trainer(max_epochs=10, gpus=-1 if torch.cuda.is_available() else None)
----> 2 trainer.fit(net, train_dataloader=data_loader)

TypeError: fit() got an unexpected keyword argument 'train_dataloader'

**Comment by jaheba** [comment]
It should be `train_dataloaders` with an `s` at the end: https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#fit

`train_dataloader` was removed here: https://github.com/Lightning-AI/lightning/pull/10325
